apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.ts

```diff
@@ -19,7 +28,95 @@ router.get(
       { assessment_id: res.locals.assessment.id },
       AssessmentAccessRulesSchema,
     );
-    res.send(InstructorAssessmentAccess({ resLocals: res.locals, accessRules }));
+
+    const assessmentPath = path.join(
+      res.locals.course.path,
+      'courseInstances',
+      res.locals.course_instance.short_name,
+      'assessments',
+      res.locals.assessment.tid,
+      'infoAssessment.json',
+    );
+
+    const assessmentPathExists = await fs.pathExists(assessmentPath);
+
+    let origHash = '';
+    if (assessmentPathExists) {
+      origHash = sha256(b64EncodeUnicode(await fs.readFile(assessmentPath, 'utf8'))).toString();
+    }
+
+    const now = Temporal.Now.zonedDateTimeISO(res.locals.course_instance.timezone);
+    const formatter = new Intl.DateTimeFormat('en-US', {
+      timeZone: res.locals.course_instance.display_timezone,
+      timeZoneName: 'short',
+    });
+    const timezone = formatter
+      .formatToParts(now.toInstant().epochMilliseconds)
+      .find((part) => part.type === 'timeZoneName')?.value;
+
+    res.send(
+      InstructorAssessmentAccess({
+        resLocals: res.locals,
+        accessRules,
+        origHash,
+        timezone: timezone ?? 'UTC',
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be an instructor)');
+    }
+
+    if (res.locals.course.example_course) {
+      throw new error.HttpStatusError(403, 'Access denied. Cannot make changes to example course.');
+    }
+
+    if (req.body.__action === 'edit_access_rules') {
+      const assessmentPath = path.join(
+        res.locals.course.path,
+        'courseInstances',
+        res.locals.course_instance.short_name,
+        'assessments',
+        res.locals.assessment.tid,
+        'infoAssessment.json',
+      );
+
+      const paths = getPaths(req, res);
+
+      const assessmentInfo = JSON.parse(await fs.readFile(assessmentPath, 'utf8'));
+
+      const origHash = req.body.__orig_hash;
+
+      const assessmentInfoEdit = assessmentInfo;
```

Comment:
This doesn't do anything; JavaScript doesn't copy on assignments. You can mutate `assessmentInfo` directly.

Reply:
Removed in [660f81b](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/660f81bebe37d51fcef94b1535961f4abfa02d88). Now mutating `assessmentInfo` directly.